runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -5320,6 +5320,10 @@ static void jitStateLogic(J9JITConfig * jitConfig, TR::CompilationInfo * compInf
          static char *disableIdleRATCleanup = feGetEnv("TR_disableIdleRATCleanup");
          if (disableIdleRATCleanup == NULL)
             persistentInfo->getRuntimeAssumptionTable()->reclaimMarkedAssumptionsFromRAT(-1);
+
+#if defined(J9VM_OPT_JITSERVER)
+         JITServer::MessageBuffer::tryFreePersistentAllocator();
```

Comment:
I am not sure whether there could be a few compilations queued when we enter idle. If that's the case, then I would place another test on compilation queue size being 0.

Reply:
Fixed.